### PR TITLE
Shaked/add style

### PR DESCRIPTION
**Pull request by shakederin**

### CODE REVIEW

**Comment by YardenPorat** [comment]
conventional commit + pr desc

File: src/components/comment/Comment.tsx [hunk 1]
```diff
@@ -14,10 +16,10 @@ const Comment = ({
     LikesCount,
     likeComment,
 }: IComment) => (
-    <li>
-        <div>
+    <li className={classes.comment}>
```
Comment (YardenPorat) [inline src/components/comment/Comment.tsx]: name the top level class for a component `root`
this is a convention we use in wcs

File: src/components/comment/Comment.tsx [hunk 2]
```diff
@@ -3,6 +3,8 @@ import CommentInfo from '../commentInfo/CommentInfo';
 import ProfilePicture from '../profilePicture/ProfilePicture';
 import CommentContent from '../commentContent/CommentContent';
 import LikeCommentButton from '../likeCommentButoon/LikeCommentButton';
+import { classes as generalClasses } from '../../style/general.st.css';
+import { classes } from './Comment.st.css';
 import type { Comment as CommentType } from '../../types';
 
 interface IComment extends CommentType {}
```
Comment (YardenPorat) [inline src/components/comment/Comment.tsx:9]: whats this?

File: src/components/comment/Comment.tsx [hunk 3]
```diff
@@ -14,10 +16,10 @@ const Comment = ({
     LikesCount,
     likeComment,
 }: IComment) => (
-    <li>
-        <div>
+    <li className={classes.comment}>
+        <div className={generalClasses.postTitle}>
```
Comment (YardenPorat) [inline src/components/comment/Comment.tsx]: post title is not very general - are you sure it belongs to general classes?

File: src/components/comment/Comment.tsx (hunk 1 above)
Comment (YardenPorat) [reply src/components/comment/Comment.tsx]: for all components

File: src/components/commentInfo/CommentInfo.tsx [hunk 4]
```diff
@@ -6,9 +8,13 @@ interface ICommentInfo {
 const CommentInfo = ({ creationDate, likesCount }: ICommentInfo) => {
     return (
         <div>
```
Comment (YardenPorat) [inline src/components/commentInfo/CommentInfo.tsx]: add root here as well, even if its empty

File: src/components/commentsList/CommentsList.tsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+import { classes } from './CommentsList.st.css';
 import type { Comment as IComment } from '../../types';
 import Comment from '../comment/Comment';
```
Comment (YardenPorat) [inline src/components/commentsList/CommentsList.tsx]: fix import order

File: src/components/imageContainer/ImageContainer.st.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.rightArrow{
+  transform: rotate(180deg);
+}
+
+.arrow{
```
Comment (YardenPorat) [inline src/components/imageContainer/ImageContainer.st.css]: so i guess this is a left arrow?

File: src/components/imageContainer/ImageContainer.st.css [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  height: 100%;
+  width: 50em;
+  background-size: cover;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.rightArrow{
+  transform: rotate(180deg);
```
Comment (YardenPorat) [inline src/components/imageContainer/ImageContainer.st.css]: `.rotateClockwise {}` ?

Comment (YardenPorat) [reply src/components/imageContainer/ImageContainer.st.css]: it doesn't really matter the only thing that I don't like is that it has nothing to do with arrows

File: src/components/inputForm/InputForm.st.css [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+  color: rgb(165, 212, 235);
+  cursor: pointer;
+}
+
+.inputForm {
+  display: flex;
+  justify-content: space-between;
+  border-top: 1px solid rgba(0, 0, 0, 0.1);
+  height: 30%;
+  display: flex;
```
Comment (YardenPorat) [inline src/components/inputForm/InputForm.st.css]: duplicate

File: src/components/inputForm/InputForm.st.css [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+.inputArea {
+  display: flex;
+  width: 100%;
+  align-items: center;
+  margin-left: 0.5em;
+}
+
+.commentInput {
+  all: unset;
```
Comment (YardenPorat) [inline src/components/inputForm/InputForm.st.css]: why?
dangerous

File: src/components/inputForm/InputForm.st.css [hunk 10]
```diff
@@ -0,0 +1,35 @@
+.postComment {
+  all: unset;
```
Comment (YardenPorat) [inline src/components/inputForm/InputForm.st.css]: avoid all unset unless there is a very good reason

File: src/components/commentsList/CommentsList.st.css [hunk 11]
```diff
@@ -0,0 +1,15 @@
+.commentsList {
+  all: unset;
```
Comment (YardenPorat) [inline src/components/commentsList/CommentsList.st.css]: avoid all unset unless there is a very good reason

File: src/components/inputForm/InputForm.tsx [hunk 12]
```diff
@@ -1,17 +1,22 @@
...
 import Icon from '../icon/Icon';
+import smileSVG from '../../assets/smile.svg';
+import { st, classes as generalClasses } from '../../style/general.st.css';
+import { classes } from './InputForm.st.css';
 
 interface IInputForm {
     onSubmit: (value: string) => void;
 }
 
 const InputForm = ({ onSubmit }: IInputForm) => (
```
Comment (YardenPorat) [inline src/components/inputForm/InputForm.tsx]: where is it used?

Comment (YardenPorat) [reply src/components/inputForm/InputForm.tsx]: onSubmit

File: src/components/inputForm/InputForm.tsx [hunk 13]
```diff
@@ -1,17 +1,22 @@
...
+            <Icon className={st(generalClasses.svgImg, classes.smileSVG)} img={smileSVG} />
+            <input className={classes.commentInput} type="text" placeholder={PLACE_HOLDER} />
         </div>
-        <button type="submit">Post</button>
+        <button className={classes.postComment} type="submit">
+            Post
+        </button>
     </form>
 );
 export default InputForm;
```
Comment (YardenPorat) [inline src/components/inputForm/InputForm.tsx:22]: on typescript, the convention is to use named exports and imports. convert everything

File: src/components/likeCommentButoon/LikeCommentButton.tsx [hunk 14]
```diff
@@ -1,11 +1,13 @@
+import heart from '../../assets/heart.svg';
+import redHeart from '../../assets/red-heart.svg';
 import Icon from '../icon/Icon';
 interface ILikeCommentButton {
     likeComment: () => void;
 }
 const LikeCommentButton = ({ likeComment }: ILikeCommentButton) => {
```
Comment (YardenPorat) [inline src/components/likeCommentButoon/LikeCommentButton.tsx]: where is `likeComment` used?

File: src/components/likeCommentButoon/LikeCommentButton.tsx [hunk 15]
```diff
@@ -1,11 +1,13 @@
...
     return (
         <div>
-            <Icon img={'heart'} className={''} />;
+            <Icon img={heart} className={''} />;
```
Comment (YardenPorat) [inline src/components/likeCommentButoon/LikeCommentButton.tsx]: `className={''}` why?

File: src/components/likeContainer/LikeContainer.tsx [hunk 16]
```diff
@@ -2,12 +2,13 @@ import { useState } from 'react';
 import type { User } from '../../types';
```
Comment (YardenPorat) [inline src/components/likeContainer/LikeContainer.tsx]: import order

File: src/components/likeContainer/LikeContainer.tsx [hunk 17]
```diff
@@ -2,12 +2,13 @@ import { useState } from 'react';
...
 import LikedByContainer from '../likedByContainer/LikedByContainer';
 import PostActionBar from '../postActionBar/PostActionBar';
+import { classes } from './LikeContainer.st.css';
 
 const LikeContainer = () => {
     const [usersWhoLiked, setUsersWhoLiked] = useState<User[]>([]);
     const likePost = () => {}; // unshift you to the state
```
Comment (YardenPorat) [inline src/components/likeContainer/LikeContainer.tsx]: what's this?

File: src/components/likeContainer/LikeContainer.tsx [hunk 18]
```diff
@@ -2,12 +2,13 @@ import { useState } from 'react';
...
     return (
-        <div>
+        <div className={classes.likeContainer}>
             <PostActionBar likePost={() => likePost()} />
```
Comment (YardenPorat) [inline src/components/likeContainer/LikeContainer.tsx]: likePost={likePost}

File: src/components/likedByContainer/LikedByContainer.tsx [hunk 19]
```diff
@@ -9,32 +11,43 @@ interface ILikedByContainer {
...
+        <div className={classes.likeByWho}>
+            <div className={classes.UserLikeByPic}>
                 <img
+                    className={st(
+                        classes.likeProfilePicture,
+                        classes.otherUserLikeByPic,
+                        generalClasses.profilePicture
+                    )}
+                    src={SAMPLE1_LIKED_BY_PROFILE_PICTURE}
+                    alt="profile Picture"
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.tsx]: inconsistent casing

File: src/components/likedByContainer/LikedByContainer.tsx [hunk 20]
```diff
@@ -9,32 +11,43 @@ interface ILikedByContainer {
...
+                ></img>
+                <img
+                    className={st(classes.likeProfilePicture, generalClasses.profilePicture)}
                     src={SAMPLE2_LIKED_BY_PROFILE_PICTURE}
                     alt="profile Picture"
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.tsx]: here as well. check other places too

File: src/components/likedByContainer/LikedByContainer.tsx [hunk 21]
```diff
@@ -9,32 +11,43 @@ interface ILikedByContainer {
...
+                        classes.likeProfilePicture,
+                        classes.otherUserLikeByPic,
+                        generalClasses.profilePicture
+                    )}
+                    src={SAMPLE1_LIKED_BY_PROFILE_PICTURE}
+                    alt="profile Picture"
+                ></img>
+                <img
+                    className={st(classes.likeProfilePicture, generalClasses.profilePicture)}
                     src={SAMPLE2_LIKED_BY_PROFILE_PICTURE}
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.tsx]: it is better to place your src's in an array than naming it SAMPLE123

File: src/components/likedByContainer/LikedByContainer.tsx [hunk 22]
```diff
@@ -9,32 +11,43 @@ interface ILikedByContainer {
...
                             {allUsersWhoLiked[0].userName}
-                            <span>and</span>
-                            <span>{allUsersWhoLiked.length - 1} others</span>
+                            <span className={generalClasses.text}>and</span>
+                            <span className={st(generalClasses.boldText, generalClasses.text)}>
+                                {allUsersWhoLiked.length - 1} others
+                            </span>
                         </>
                     ) : (
                         'yonni'
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.tsx]: consider moving all your example texts to a single file

File: src/components/postActionBar/PostActionBar.tsx [hunk 23]
```diff
@@ -1,14 +1,20 @@
...
     likePost: () => void;
 }
 
 const PostActionBar = ({ likePost }: IPostActionBar) => (
-    <div>
+    <div className={classes.icons}>
         <div>
-            <Icon img={'heart'} className={''} />
+            <Icon img={heart} className={generalClasses.svgImg} />
             <svg
```
Comment (YardenPorat) [inline src/components/postActionBar/PostActionBar.tsx]: consider moving svg to its own component

File: src/components/postDescription/PostDescription.st.css [hunk 24]
```diff
@@ -0,0 +1,24 @@
...
+  justify-content: space-between;
+  align-items: flex-start;
+  padding-left: 0.3em;
+
+}
+.commentText {
+  margin: 0;
+  padding: 0em;
+  font-weight: 400;
+  margin-left: 1em;
```
Comment (YardenPorat) [inline src/components/postDescription/PostDescription.st.css]: double margin

File: src/components/postDescription/PostDescription.st.css [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+  margin: 1.5em 0em 0em 0em;
+  display: flex;
+  justify-content: space-between;
+  align-items: flex-start;
+  padding-left: 0.3em;
+
+}
+.commentText {
+  margin: 0;
+  padding: 0em;
```
Comment (YardenPorat) [inline src/components/postDescription/PostDescription.st.css]: `padding: 0;`

File: src/components/username/Username.tsx [hunk 26]
```diff
@@ -1,15 +1,18 @@
+import { classes as generalClasses } from '../../style/general.st.css';
+import { classes } from './UserName.st.css';
+
 interface IUsername {
     userName: string;
     header: boolean;
 }
 const Username = ({ userName, header }: IUsername) => (
```
Comment (YardenPorat) [inline src/components/username/Username.tsx]: sometimes you write Username and sometimes userName.
Is this one word or two?

File: src/constants.ts [hunk 27]
```diff
@@ -1,8 +1,12 @@
 export const SAMPLE_DATE = new Date();
```
Comment (YardenPorat) [inline src/constants.ts]: so I see you have a file for example data
move every here, and divide them to objects according to topic

```ts
const SAMPLE_PROFILE_PICTURES = [
   ...
]
```

Comment (YardenPorat) [reply src/constants.ts]:
```ts
EXAMPLE_TEXTS: {
    DESCRIPTION: 'bla bla',
    ...
}

File: src/style/general.st.css [hunk 28]
```diff
@@ -0,0 +1,45 @@
+:global(body) {
+  margin: 0;
+}
+
+
```
Comment (YardenPorat) [inline src/style/general.st.css:5]: remove newline

**Review by YardenPorat — CHANGES_REQUESTED** [review]

File: src/components/imageContainer/ImageContainer.st.css (hunk 6 above)
Comment (shakederin) [reply src/components/imageContainer/ImageContainer.st.css]: its a general arrow.
right arrow is extension of arrow

File: src/components/imageContainer/ImageContainer.st.css (hunk 7 above)
Comment (shakederin) [reply src/components/imageContainer/ImageContainer.st.css]: Scroll arrows?

File: src/components/inputForm/InputForm.tsx (hunk 13 above)
Comment (shakederin) [reply src/components/inputForm/InputForm.tsx:22]: on type and "free text" (Post)?

File: src/components/inputForm/InputForm.tsx (hunk 12 above)
Comment (shakederin) [reply src/components/inputForm/InputForm.tsx]: הכנה למזגן

File: src/components/inputForm/InputForm.tsx (hunk 13 above)
Comment (shakederin) [reply src/components/inputForm/InputForm.tsx:22]: ?

File: src/components/comment/Comment.tsx (hunk 2 above)
Comment (shakederin) [reply src/components/comment/Comment.tsx:9]: Make all the componnents with the same interface naming format => I + component's name.
Here type and component has the same name

File: src/components/commentsList/CommentsList.st.css (hunk 11 above)
Comment (shakederin) [reply src/components/commentsList/CommentsList.st.css]: keeping this one, will be removed after functionality

File: src/components/inputForm/InputForm.tsx (hunk 13 above)
Comment (shakederin) [reply src/components/inputForm/InputForm.tsx:22]: will solve in different PR

File: src/components/likeCommentButoon/LikeCommentButton.tsx (hunk 14 above)
Comment (shakederin) [reply src/components/likeCommentButoon/LikeCommentButton.tsx]: כנה למזגן

File: src/components/comment/Comment.tsx [hunk 29]
```diff
@@ -1,8 +1,9 @@
...
+import UserName from '../username/Username';
 import CommentInfo from '../commentInfo/CommentInfo';
 import ProfilePicture from '../profilePicture/ProfilePicture';
 import CommentContent from '../commentContent/CommentContent';
-import LikeCommentButton from '../likeCommentButoon/LikeCommentButton';
+import LikeCommentButton from '../likeCommentButton/LikeCommentButton';
+import { classes } from './Comment.st.css';
 import type { Comment as CommentType } from '../../types';
 
 interface IComment extends CommentType {}
```
Comment (YardenPorat) [inline src/components/comment/Comment.tsx:9]: interfaces for props are usually called `SomethingProps` or `SomethingOptions`

Comment (YardenPorat) [reply src/components/comment/Comment.tsx:9]: not mandatory

Comment (YardenPorat) [reply src/components/comment/Comment.tsx:9]: if you export the component - export the interface (this is mandatory)

File: src/components/imageContainer/ImageContainer.st.css (hunk 7 above)
Comment (YardenPorat) [reply src/components/imageContainer/ImageContainer.st.css]: I feel like it is a utility class that rotate stuff 180 deg

File: src/components/commentsList/CommentsList.st.css (hunk 11 above)
Comment (YardenPorat) [reply src/components/commentsList/CommentsList.st.css]: wdym?

File: src/components/inputForm/InputForm.tsx (hunk 12 above)
Comment (YardenPorat) [reply src/components/inputForm/InputForm.tsx]: comment on PRs in English only

File: src/components/inputForm/InputForm.tsx (hunk 13 above)
Comment (YardenPorat) [reply src/components/inputForm/InputForm.tsx:22]: okay

File: src/components/imageContainer/ImageContainer.st.css [hunk 30]
```diff
@@ -0,0 +1,20 @@
+.root{}
```
Comment (YardenPorat) [inline src/components/imageContainer/ImageContainer.st.css]: formatting

File: src/components/imageContainer/ImageContainer.st.css [hunk 31]
```diff
@@ -0,0 +1,20 @@
...
+.mainImg {
+  height: 100%;
+  width: 50em;
+  background-size: cover;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.rightArrow{
```
Comment (YardenPorat) [inline src/components/imageContainer/ImageContainer.st.css]: formatting

File: src/components/imageContainer/ImageContainer.st.css (hunk 6 above)
Comment (YardenPorat) [reply src/components/imageContainer/ImageContainer.st.css]: `.arrow {`

File: src/components/likedByContainer/LikedByContainer.st.css [hunk 32]
```diff
@@ -0,0 +1,31 @@
+.root {
+  margin: 1em 0em 0em 0.5em;
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.st.css:2]: `margin: 1em 0 0 0.5em;`

File: src/components/likedByContainer/LikedByContainer.st.css [hunk 33]
```diff
@@ -0,0 +1,31 @@
...
+  display: flex;
+}
+.UserLikeByPic {
+  display: flex;
+  flex-direction: row-reverse;
+}
+.likeProfilePicture {
+  width: 1.5em;
+  height: 1.5em;
+  border: 0em;
```
Comment (YardenPorat) [inline src/components/likedByContainer/LikedByContainer.st.css:16]: same

File: src/components/postDescription/PostDescription.st.css [hunk 34]
```diff
@@ -0,0 +1,28 @@
+.root{
```
Comment (YardenPorat) [inline src/components/postDescription/PostDescription.st.css:1]: formatting

File: src/components/postHeader/PostHeader.st.css [hunk 35]
```diff
@@ -0,0 +1,16 @@
+.root {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
+}
+.profilePictureHeaderBlock {
```
Comment (YardenPorat) [inline src/components/postHeader/PostHeader.st.css:8]: add newline

File: src/components/postHeader/PostHeader.st.css [hunk 36]
```diff
@@ -0,0 +1,16 @@
...
+  display: flex;
+  padding-left: 0.3em;
+  height: 4em;
+  align-items: center;
+}
+.ellipsis {
```
Comment (YardenPorat) [inline src/components/postHeader/PostHeader.st.css:15]: add newline

File: src/mockText.ts [hunk 37]
```diff
@@ -0,0 +1,16 @@
+export const MOCK_DATA = {
+    SAMPLE_DATE: new Date(),
+    SAMPLE_USER_NAME: 'tonytoscani',
+    SAMPLE_NAME: 'yonni',
+    SAMPLE_HOURS_AGO: '18 HOURS AGO',
+    SAMPLE_POST_DESCRIPTION: 'me, at the sunrise',
+    SAMPLE_TIME_PASSED_FROM_CREATION_DATE: '3 d',
+};
```
Comment (YardenPorat) [inline src/mockText.ts:8]: well done

File: src/style/general.st.css (hunk 28 above)
Comment (YardenPorat) [reply src/style/general.st.css:5]: you liked but didn't remove

File: src/style/general.st.css [hunk 38]
```diff
@@ -0,0 +1,38 @@
...
+  font-weight: 400;
+  margin-left: 0.3em;
+}
+
+.profilePicture {
+  height: 2em;
+  margin-left: 0.5em;
+  border-radius: 50%;
+  border: double 1em transparent;
+  background-image: linear-gradient(white, white),
```
Comment (YardenPorat) [inline src/style/general.st.css]: `linear-gradient(white, white)` ?????

**Review by YardenPorat — APPROVED** [review]
Fix comments and merge

File: src/style/general.st.css (hunk 28 above)
Comment (shakederin) [reply src/style/general.st.css:5]: you are right!

removed in a different branch

**Pull request merged**